src/adapter/src/optimize/peek.rs

```diff
@@ -308,6 +324,69 @@ impl<'s> Optimize<LocalMirPlan<ResolvedLocal<'s>>> for Optimizer {
             );
         }
 
+        // Adding s3 sink if initiated from copy to
+        if let PeekContext::CopyTo(copy_to_context) = peek_ctx {
```

Comment:
To me this seems like it belongs somewhere else (not sure where tho)? @aalexandrov do you have thoughts? (I don't know much about stages MIR and below though, so this could be correct). Maybe someone from storage also has a thought.

Reply:
So, this part is doing a HirScalar -> MirScalar -> String -> Uri for the copy target before passing that value to the s3 sink. Since the `LocalMirPlan` is already supposed to have Mir, maybe the Hir -> Mir could have happened earlier. But the validation upon the s3 url itself can only be done after getting the `Uri` value out of it.

Reply:
Actually let me clean this up a bit. The scalar can be evaluated earlier.

Reply:
I moved the scalar evaluation earlier. Keeping only the compute sink addition to the dataflow.